### PR TITLE
Refresh the documentation with K8S library clarification

### PR DESCRIPTION
**Pull request by potiuk**

### CODE REVIEW

**Comment by potiuk** [comment]
I am not sure why we have so many diffs - those look like different way some characters are either quoted or not..  But the gist of the change is the note clarification about the library rather than cluster :)

File: docs-archive/apache-airflow/2.3.0/release_notes.html [hunk 1]
```diff
@@ -2020,8 +2020,13 @@ <h4>Task log templates are now read from the metadata database instead of <code
...
 <p>A new <code class="docutils literal notranslate"><span class="pre">log_template</span></code> table is introduced to solve this problem. This table is synchronized with the aforementioned config values every time Airflow starts, and a new field <code class="docutils literal notranslate"><span class="pre">log_template_id</span></code> is added to every DAG run to point to the format used by tasks (<code class="docutils literal notranslate"><span class="pre">NULL</span></code> indicates the first ever entry for compatibility).</p>
 </div>
-<div class="section" id="minimum-kubernetes-version-bumped-from-3-0-0-to-21-7-0-20759">
-<h4>Minimum kubernetes version bumped from <code class="docutils literal notranslate"><span class="pre">3.0.0</span></code> to <code class="docutils literal notranslate"><span class="pre">21.7.0</span></code> (#20759)<a class="headerlink" href="#minimum-kubernetes-version-bumped-from-3-0-0-to-21-7-0-20759" title="Permalink to this headline">¶</a></h4>
+<div class="section" id="minimum-kubernetes-library-version-bumped-from-3-0-0-to-21-7-0-20759">
```
Comment (potiuk) [inline docs-archive/apache-airflow/2.3.0/release_notes.html:2023]: This is the "actual change :)

**Comment by potiuk** [comment]
chance for merging this one @kaxil @jedcunningham @dstandish -> already in main and marked as 2.1.0 but would be nice to get it to the docs (we already had one user mislead that "kubernetes" means "kubernetes cluster" rather than library.

**Comment by potiuk** [comment]
Anyone :) ?

**Review by kaxil — APPROVED** [review]

**Pull request merged**